### PR TITLE
Adding tweets crawl

### PR DESCRIPTION
**Pull request by SCcagg5**
adding crawl google news (overriding other branch :/)
adding crawl tweet
adding Dockerfile version 3.7 and 3.8

adding requirement (snscrape && pandas)

fixing version used by services

### CODE REVIEW

File: flow/scripts/crawl_tweet.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    helpers.bulk(es_inst, actions)
+
+class GetTweets(Task):
+    def run(self, index_name):
+        d = datetime.now() - timedelta(days=1)
+        limit = d.strftime("%y-%m-%d")
+        to_inject = []
+        for i,tweet in enumerate(sntwitter.TwitterSearchScraper("covid since:" + limit).get_items()):
+            if i>300:
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py]: Make this limit a global variable at top of script

(maybe we should lower it to 100 for example too)

File: flow/scripts/crawl_tweet.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            if response['acknowledged'] == True:
+                logger.info("INDEX MAPPING SUCCESS FOR INDEX: {}".format(response['index']))
+            elif 'error' in response:
+                logger.error(response['error']['root_cause'])
+                logger.error("Error type: {}".format(response['error']['type']))
+                raise Exception("Unable to create index mapping")
+
+
+if __name__ == "__main__":
+    with Flow("Search tweets and insert") as flow:
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py]: Set this flow description the same as `project_name`

File: flow/scripts/crawl_tweet.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            keyword_tasks=dict(index_name=index))
+
+    try:
+        client = Client()
+        client.create_project(project_name=name)
+    except prefect.utilities.exceptions.ClientError as e:
+        logger.info("Project already exists")
+
+    flow.register(project_name=name, labels=["development"])
+    flow.run()
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py]: Remove this line (should only be executed in prod by Prefect)

File: flow/scripts/crawl_tweet.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+if __name__ == "__main__":
+    with Flow("Search tweets and insert") as flow:
+        flow.set_dependencies(
+            task=GetTweets(),
+            upstream_tasks=[GenerateEsMapping(index)],
+            keyword_tasks=dict(index_name=index))
+
+    try:
+        client = Client()
+        client.create_project(project_name=name)
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py]: The global variable `name` should probably be `project_name`

File: flow/scripts/crawl_tweet.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    except prefect.utilities.exceptions.ClientError as e:
+        logger.info("Project already exists")
+
+    flow.register(project_name=name, labels=["development"])
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py]: Add `add_default_labels=False`

```
flow.register(
        project_name=project_name, labels=["development"], add_default_labels=False
    )
```

**Review by flavienbwk — CHANGES_REQUESTED** [review]

File: flow/scripts/crawl_google_news.py [hunk 6]
```diff
@@ -1,12 +1,14 @@
-# python3
+#python3
```
Comment (flavienbwk) [inline flow/scripts/crawl_google_news.py]: This script should be the same as the one on develop

**Review by flavienbwk — CHANGES_REQUESTED** [review]

**Comment by SCcagg5** [comment]
Same format of code in every crawler -> customisable variables on top of script and schedule

**Review by flavienbwk — APPROVED** [review]

**Review by flavienbwk — COMMENTED** [review]
Going to change mapping to be compatible with GoogleNews

**Pull request merged**